src/App.jsx

```diff
@@ -0,0 +1,443 @@
+    }
+  }, [token,currentWorkspace]); //only fetch tasks and workspace belonging to user
+
+  //Authentication Endpoints
+
+  //Login
+  const handleLogin = async () => {
+    setLoginError(""); //Reset error before new attempt
+    try {
+      const response = await fetch(`${API_BASE}/login`, { //api call
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ email, password })
+      });
+
+      const data = await response.json(); //upon receiving api call apply data + set status to login (unless error)
+      if (response.ok) {
+        localStorage.setItem("token", data.token);
+        localStorage.setItem("userEmail", email);
+        setToken(data.token);
+        setUserEmail(email);
+        setEmail("");
+        setPassword("");
+        console.log("Login successful!");
+      } else {
+        setLoginError(data.message || "Invalid credentials");
+        console.error("Login failed:", data.message);
+      }
+    } catch (error) {
+      setLoginError("Network error. Please try again.");
+      console.error("Error:", error);
+    }
+  };
+
+  //Logout
+
+  const handleLogout = () => { //take away the login token
+    localStorage.removeItem("token");
+    localStorage.removeItem("userEmail");
+    setToken("");
+    setUserEmail("");
+    console.log("Logged out");
+  };
+
+  //SignUp (createUser)
+
+  const createUser = async () => {
+    try {
+      const response = await fetch(`${API_BASE}/new_user`, { //api call
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          username: "TestUser",
+          email: email,
+          password: password,
+          name: "Test Name"
+        })
+      });
+
+      if (!response.ok) throw new Error("Failed to create user");
+      console.log("User created successfully");
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  };
+
+  //Task Endpoints
+  
+  const fetchTasks = async () => {
+    try {
+      let url = `${API_BASE}/tasks`;
+      if (currentWorkspace) {
+        url += `?workspaceId=${currentWorkspace.id}`;
+      }
+    
+      const response = await fetch(url, {  
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+          "Authorization": `Bearer ${token}`,
+        },
+      });
+    
+      if (!response.ok) throw new Error(`Server responded with ${response.status}`);
+    
+      const data = await response.json();
+      console.log("Fetched tasks:", data.tasks);
+    
+      if (!Array.isArray(data.tasks)) {
+        console.error("Error: tasks is not an array", data);
+        setTasks([]); 
+        return;
+      }
+    
+      //Ensure array
+      const formattedTasks = data.tasks.map((task) => {
+        const tagsArray = Array.isArray(task.tags)
+          ? task.tags
+          : task.tags
+          ? task.tags.split(",").map(tag => tag.trim())
+          : [];
```

Comment:
I think this might be a little more readable using optional chaining / null coalescing

```suggestion
        const tagsArray = Array.isArray(task.tags)
          ? task.tags
          : task.tags?.split(",").map(tag => tag.trim()) ?? [];
```

though the backend should definitely be returning a proper list of tags (which will probably each look something like `{id: ..., name: 'some-tag'}`) so this may be a non-issue anyway